### PR TITLE
compatible with the old version and take effect of dispatch and causality

### LINKED ISSUE #908: drainer flag compatible with the old version
## Feature Request

**Is your feature request related to a problem? Please describe:**

In the latest binlog  #785 , the startup flags of ~~pump and~~ drainer are incompatible with the old version. For example, disable-detect is changed to enable-detect and disable-dispatch is changed to enable-dispatch.
**Describe the feature you'd like:**

Colleagues can use disable-XXX and enable-XXX. When both exist, disable-XXX takes effect first.
**Describe alternatives you've considered:**

Colleagues can use disable-XXX and enable-XXX. When both exist, disable-XXX takes effect first.
**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by july2993**
### What problem does this PR solve? 
- fix #908 compatible with the `disable-detect` and `disable-causality`
- make the two flags take effect truly
- make the `merge` option in `Loader` as default (the previous effect is false actually)
### What is changed and how it works?
- add more field for compatible the disable* config, if the old one(disable one) is set, the old one takes effect.
- add option `EnableDispatch` and `EnableCausality` for `pkg/Loader`

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

 - Need to be included in the release note

### CODE REVIEW

File: drainer/config.go [hunk 1]
```diff
@@ -75,9 +75,62 @@ type SyncerConfig struct {
...
 	DestDBType        string             `toml:"db-type" json:"db-type"`
 	Relay             RelayConfig        `toml:"relay" json:"relay"`
-	EnableDispatch    bool               `toml:"enable-dispatch" json:"enable-dispatch"`
-	SafeMode          bool               `toml:"safe-mode" json:"safe-mode"`
-	EnableCausality   bool               `toml:"enable-detect" json:"enable-detect"`
+	// disable* is keep for backward compatibility.
+	// if both setted, the disable one take affect.
+	DisableDispatchFlag *bool `toml:"disable-dispatch-flag" json:"disable-dispatch-flag"`
+	EnableDispatchFlag  *bool `toml:"enable-dispatch-flag" json:"enable-dispatch-flag"`
+	DisableDispatchFile *bool `toml:"disable-dispatch" json:"disable-dispatch"`
```
Comment (WangXiangUSTC) [inline drainer/config.go]: how about use these configs in integration test

Comment (july2993) [reply drainer/config.go]: add in https://github.com/pingcap/tidb-binlog/pull/915/commits/505451b4944d87c9d42b833e939458de0903d40f

**Comment by WangXiangUSTC** [comment]
LGTM

**Comment by july2993** [comment]
@GregoryIan PTAL

**Comment by july2993** [comment]
/run-unit-test  
the unstable test will fixed in https://github.com/pingcap/tidb-binlog/pull/918

**Comment by GregoryIan** [comment]
@csuzhangxc PTAL

**Comment by csuzhangxc** [comment]
I'll task a look later today.

File: pkg/loader/load.go [hunk 2]
```diff
@@ -184,20 +209,28 @@ func NewLoader(db *gosql.DB, opt ...Option) (Loader, error) {
 		o(&opts)
 	}
 
+	log.Info("new loader", zap.String("opts", fmt.Sprintf("%+v", opts)))
+
+	if !opts.enableDispatch {
+		opts.workerCount = 1
+		opts.batchSize = math.MaxInt64
```
Comment (csuzhangxc) [inline pkg/loader/load.go]: Why do we change this behavior (to `math.MaxInt64`)?

File: drainer/config.go [hunk 3]
```diff
@@ -75,9 +75,62 @@ type SyncerConfig struct {
 	DoDBs             []string           `toml:"replicate-do-db" json:"replicate-do-db"`
 	DestDBType        string             `toml:"db-type" json:"db-type"`
 	Relay             RelayConfig        `toml:"relay" json:"relay"`
-	EnableDispatch    bool               `toml:"enable-dispatch" json:"enable-dispatch"`
-	SafeMode          bool               `toml:"safe-mode" json:"safe-mode"`
-	EnableCausality   bool               `toml:"enable-detect" json:"enable-detect"`
+	// disable* is keep for backward compatibility.
+	// if both setted, the disable one take affect.
+	DisableDispatchFlag *bool `toml:"disable-dispatch-flag" json:"disable-dispatch-flag"`
+	EnableDispatchFlag  *bool `toml:"enable-dispatch-flag" json:"enable-dispatch-flag"`
```
[the remark this answers is not in the document]
Comment (july2993) [reply drainer/config.go]: done https://github.com/pingcap/tidb-binlog/pull/915/commits/aeb33cda5c59d96b3c07adf8bc9835c088d1860f

File: pkg/loader/load.go (hunk 2 above)
Comment (july2993) [reply pkg/loader/load.go]: add comment in https://github.com/pingcap/tidb-binlog/pull/915/commits/aeb33cda5c59d96b3c07adf8bc9835c088d1860f

**Pull request merged**